### ISSUE TITLE
TPM failure when running TPM2_PCR_Event through tpm2-abrmd

### ISSUE THREAD

**Issue by diabonas** [stefanberger/libtpms#259]
**Describe the bug**
Since libtpms version 0.8.5, attempting to run `tpm2 pcrevent` (from [tpm2-tools](https://github.com/tpm2-software/tpm2-tools)) through [tpm2-abrmd](https://github.com/tpm2-software/tpm2-abrmd) generates the following TPM failure:
```
libtpms/tpm2: Entering failure mode; code: 3, location: CryptDigestUpdate line 429
libtpms/tpm2: TPM2_Process: Entered failure mode through command:
80 02 00 00 00 22 00 00 01 5c 80 00 00 00 00 00
00 09 40 00 00 09 00 00 00 00 00 00 05 74 65 73
74 0a
```
I guess this is related to commit https://github.com/stefanberger/libtpms/commit/33a03986e0a09dde439985e0312d1c8fb3743aab, which seems to corrupt the internal TPM state in this use case.

**To Reproduce**
1. Start swtpm and tpm2-abrmd:
    ```
    swtpm socket --tpm2 --server port=2321 --ctrl type=tcp,port=2322 --flags not-need-init --tpmstate dir=/tmp &
    tpm2-abrmd --session --dbus-name=com.intel.tss2.Tabrmd2321 --tcti=swtpm:port=2321 &
    ```
2. Try to run `tpm2 pcrevent` using the simulator:
    ```
    $ echo test | tpm2 pcrevent --tcti="tabrmd:bus_type=session,bus_name=com.intel.tss2.Tabrmd2321"
    WARNING:esys:src/tss2-esys/api/Esys_SequenceUpdate.c:287:Esys_SequenceUpdate_Finish() Received TPM Error
    ERROR:esys:src/tss2-esys/api/Esys_SequenceUpdate.c:97:Esys_SequenceUpdate() Esys Finish ErrorCode (0x00000101)
    ERROR: Esys_SequenceUpdate(0x101) - tpm:error(2.0): commands not being accepted because of a TPM failure
    WARNING:esys:src/tss2-esys/api/Esys_FlushContext.c:234:Esys_FlushContext_Finish() Received TPM Error
    ERROR:esys:src/tss2-esys/api/Esys_FlushContext.c:89:Esys_FlushContext() Esys Finish ErrorCode (0x00000101)
    ERROR: Esys_FlushContext(0x101) - tpm:error(2.0): commands not being accepted because of a TPM failure
    ERROR: Unable to run pcrevent
    ```

**Expected behavior**
The command should succeed and print the calculated hashes. Instead, libtpms/swtpm enters a TPM failure with the above error messages.

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Version: rolling release

**Versions of relevant components**
- libtpms: 0.8.5
- openssl: 1.1.1l
- tpm2-tss: 3.1.0
- tpm2-abrmd: 2.4.0
- tpm2-tools: 5.1.1

**Additional context**
When running `tpm2 pcrevent` directly against the simulator instead of through the tpm2-abrmd context manager (i.e. with `echo test | tpm2 pcrevent --tcti=swtpm`), the problem doesn't occur, probably because there is no context save involved.

**Comment by stefanberger** [comment]
Thanks for the report. The issue is that event sequence objects were never context-saved properly. I will create v0.8.6 soon.

**Comment by stefanberger** [comment]
With the patch applied multiple of these can now be run in parallel:
```
dd bs=1024 count=1024 if=/dev/zero status=none | \
  tpm2 pcrevent --tcti="tabrmd:bus_type=session,bus_name=com.intel.tss2.Tabrmd2321"
```

**Comment by stefanberger** [comment]
@diabonas I let you close the issue. It should be fixed now on master and stable-0.8.

**Comment by diabonas** [comment]
Done, thanks again!
